google/cloud/spanner_dbapi/cursor.py

```diff
@@ -126,30 +147,31 @@ def rowcount(self):
             stacklevel=2,
         )
 
-    def _raise_if_closed(self):
-        """Raise an exception if this cursor is closed.
+    @check_not_closed
+    def callproc(self, procname, args=None):
+        """A no-op, raising an error if the cursor or connection is closed."""
+        pass
 
-        Helper to check this cursor's state before running a
-        SQL/DDL/DML query. If the parent connection is
-        already closed it also raises an error.
+    @check_not_closed
+    def nextset(self):
+        """A no-op, raising an error if the cursor or connection is closed."""
+        pass
 
-        :raises: :class:`InterfaceError` if this cursor is closed.
-        """
-        if self.is_closed:
-            raise InterfaceError("Cursor and/or connection is already closed.")
+    @check_not_closed
+    def setinputsizes(self, sizes):
+        """A no-op, raising an error if the cursor or connection is closed."""
+        pass
 
-    def callproc(self, procname, args=None):
+    @check_not_closed
+    def setoutputsize(self, size, column=None):
         """A no-op, raising an error if the cursor or connection is closed."""
-        self._raise_if_closed()
+        pass
 
     def close(self):
         """Closes this cursor."""
         self._is_closed = True
 
     def _do_execute_update(self, transaction, sql, params):
-        sql = parse_utils.ensure_where_clause(sql)
-        sql, params = parse_utils.sql_pyformat_args_to_spanner(sql, params)
```

Comment:
There are the same lines earlier, but under an `if` statement. Merging both cases into one.